pypesto/collections/collections.py

```diff
@@ -0,0 +1,484 @@
+        CollectionPrediction object.
+
+        Parameters
+        ----------
+        percentiles_list:
+            List or tuple of percent numbers for the percentiles
+
+        Returns
+        -------
+        summary:
+            Dict with mean, std, median, and percentiles of parameter vectors
+        """
+        # compute summaries based on parameters
+        summary = {MEAN: np.mean(self.x_vectors, axis=1),
+                   STANDARD_DEVIATION: np.std(self.x_vectors, axis=1),
+                   MEDIAN: np.median(self.x_vectors, axis=1)}
+        for perc in percentiles_list:
+            summary[f'{PERCENTILE} {perc}'] = np.percentile(self.x_vectors,
+                                                            perc, axis=1)
+        # store and return results
+        self.summary = summary
+        return summary
+
+    def check_identifiability(self) -> pd.DataFrame:
+        """
+        This functions uses ensemble mean and standard deviation to assess
+        (in a rudimentary way) whether or not parameters are identifiable.
+        It returns a dataframe with tuples, which specify whether or not the
+        lower and the upper bound are violated
+
+        Returns
+        -------
+        parameter_identifiability:
+            DataFrame indicating parameter identifiability based on mean
+            plus/minus standard deviations and parameter bounds
+        """
+        # first check if summary was computed. If not, do so
+        if self.summary is None:
+            self.compute_summary()
+
+        # check identifiability for each parameter
+        parameter_identifiability = []
+        for ix, x_name in enumerate(self.x_names):
+            # define some short hands
+            lb = self.lower_bound[ix]
+            ub = self.upper_bound[ix]
+            mean = self.summary[MEAN][ix]
+            std = self.summary[STANDARD_DEVIATION][ix]
+            median = self.summary[MEAN][ix]
+
+            # create dict of identifiability
+            parameter_identifiability.append({
+                'parameterId': x_name,
+                'lowerBound': lb,
+                'upperBound': ub,
+                'collection_mean': mean,
+                'collection_std': std,
+                'collection_median': median,
+                'within lb: 1 std': lb < mean - std,
+                'within ub: 1 std': ub > mean + std,
+                'within lb: 2 std': lb < mean - 2 * std,
+                'within ub: 2 std': ub > mean + 2 * std,
+                'within lb: 3 std': lb < mean - 3 * std,
+                'within ub: 3 std': ub > mean + 3 * std,
+            })
+
+        # create DataFrame
+        parameter_identifiability = pd.DataFrame(parameter_identifiability)
+        parameter_identifiability.index = \
+            parameter_identifiability['parameterId']
+
+        return parameter_identifiability
+
+
+def read_from_csv(path: str,
+                  sep: str = '\t',
```

Comment:
Seems odd (as in the previous `Prediction` PR) to call this a `csv` method, since it defaults to `tsv`.

Reply:
in practice, we will always want to use tsv, I think. But for some reason, csv is the most common term for a file formatted like this... Just as Pandas also calls its function `read_csv` and `to_csv` ... I just used an analogous naming...

Reply:
Agreed but e.g. pandas `read_csv` defaults to `read_csv(..., sep=',')`, not `read_csv(..., sep='\t')`. Not so important but may confuse users who are used to CSV meaning comma by default.